Templates/Templates/AutoDecodable.stencil

```diff
@@ -0,0 +1,96 @@
+{% macro keyedNameFor type  variable %}{% if variable|annotated:"codingKey" %} = "{{ variable.annotations.codingKey }}"{% elif type|annotated:"codingKeyTransform" %}= "{% call decodeNameWithTransform variable.name type.annotations.codingKeyTransform %}"{% endif %}{% endmacro %}
+
+{# the keys to use to decoding #}
+{% macro codingKeys type %}
+  private enum CodableKeys: String, CodingKey {
+    {% for variable in type.instanceVariables where variable|!annotated:"skipCoding" %}
+    case {{ variable.name }} {% call keyedNameFor type variable %}
+    {% endfor %}
+  }
+{% endmacro %}
+
+{% macro inheritanceInitializer type %}
+  {% if type.supertype.based.AutoDecodable or type.supertype.based.Decodable %}
+    try self.init(from: decoder)
+  {% else %}
+    self.init()
+  {% endif %}
+{% endmacro %}
+
+
+{% macro convertTypeForEncoder var %}
+  {% set encodeCall %}{% if var.isOptional %}encodeIfPresent{% else %}encode{% endif %}{% endset %}
+  {% if var|!annotated:"encodingType" %}
+    try container.{{ encodeCall }}({{ var.name }}, forKey: .{{ var.name }})
+  {% else %}
+  {% set typename %}{{ var.typeName.name }}{% endset %}
+  {% set encodingType %}{{ var.annotations.encodingType }}{% endset %}
+  {% if typename == "Bool" and encodingType == "String" %}
+    try container.encode({{ var.name }} ? 'true' : 'false', forKey: .{{ var.name }})
+  {% elif typename == "Bool" and encodingType == "Int" %}
+    try container.encode({{ var.name }} ? 1 : 0, forKey: .{{ var.name }})
+  {% endif %}
+  {% endif %}
+{% endmacro %}
+
+{% macro encode type %}
+  {{ type.accessLevel }} {% if type.supertype.implements.Encodable or type.supertype.implements.AutoEncodable %}override {% endif %}func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodableKeys.self)
+    {% for variable in type.instanceVariables where variable|!annotated:"skipCoding" %}
+    {% call convertTypeForEncoder variable %}
+    {% endfor %}
+  }
+{% endmacro %}
+
+{% for type in types.all|!class where type.based.AutoDecodable or type|annotated:"AutoDecodable" and type.name != "AutoCodable" and type.name != "AutoDecodable" %}
+// MARK: {{ type.name }}+Decodable | AutoDecodable
+extension {{ type.name }} {
+  {% call codingKeys type %}
+
+  {{ type.accessLevel }} init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodableKeys.self)
+    {% for var in type.instanceVariables where var|!annotated:"skipCoding" %}
+    {% if var|annotated:"codingDefaultValue" or var.isOptional %}
+    {{ var.name }} = try container.decodeOrDefault({{ var.typeName.name }}.self, forKey: .{{ var.name }}, defaultValue: {% call defaultValueFor var %}
+    {% else %}
+    {{ var.name }} = try container.decode({{ var.typeName.name }}.self, forKey: .{{ var.name }})
+    {% endif %}
+    {% endfor %}
+  }
+}
+{% endfor %}
+
+{# todo inheritance #}
```

Comment:
I think it should be completely removed from template until it is implemented fully or at least up to some extent.